lineage/BaumWelch.py

```diff
@@ -73,44 +73,42 @@ def fit(tHMMobj, tolerance=np.spacing(1), max_iter=200):
 
     # first stopping condition check
     new_LL = calculate_log_likelihood(NF)
-    for _ in range(max_iter):
+    for iter_number in range(max_iter):
         old_LL = new_LL
 
         # code for grouping all states in cell lineages
-        cell_groups = [[] for state in range(num_states)]
         pi_estimate = np.zeros((num_states), dtype=float)
-        T_estimate = np.zeros((num_states, num_states), dtype=float)
+        numer_estimate = np.zeros((num_states, num_states), dtype=float)
+        denom_estimate = np.zeros((num_states,),dtype=float)
         for num, lineageObj in enumerate(tHMMobj.X):
+            
             lineage = lineageObj.output_lineage
             gamma_array = gammas[num]
+            
+            # local pi estimate
             pi_estimate += gamma_array[0, :]
 
-            denom = get_all_gammas(lineageObj, gamma_array)
-            numer = get_all_zetas(
+            # local T estimate
+            numer_estimate += get_all_zetas(
                 lineageObj=lineageObj, beta_array=betas[num], MSD_array=tHMMobj.MSD[num], gamma_array=gamma_array, T=tHMMobj.estimate.T
             )
+            denom_estimate += get_all_gammas(lineageObj, gamma_array)
```

Comment:
The values are collected and summed over the lineages first, before their ratio is taken